ql/experimental/math/fireflyalgorithm.cpp

```diff
@@ -178,22 +185,28 @@ namespace QuantLib {
                     //Loop over dimensions
                     for (Size j = 0; j < N_; j++) {
                         //Update position
-                        x[j] += xI[j] + xRW[j];
+                        z[j] = x[j] + xI[j] + xRW[j];
                         //Enforce bounds on positions
-                        if (x[j] < lX_[j]) {
-                            x[j] = lX_[j];
+                        if (z[j] < lX_[j]) {
+                            z[j] = lX_[j];
                         }
-                        else if (x[j] > uX_[j]) {
-                            x[j] = uX_[j];
+                        else if (z[j] > uX_[j]) {
+                            z[j] = uX_[j];
                         }
                     }
-                    //Evaluate x & mark best
-                    values_[index].first = P.value(x);
-                    if (values_[index].first < bestValue) {
-                        bestValue = values_[index].first;
-                        bestX = x;
-                        iterationStat = 0;
-                    }
+                    Real val = P.value(z);
+                    if(!std::isnan(val))
```

Comment:
`std::isnan` is only available since C++11 and is not supported by older compilers.
Is it really necessary here, or was it just a precaution?